### PR TITLE
Make GHC 8.6.1 the default one

### LINKED ISSUE #151: Make GHC 8.6.3 the default one for the generated project
Also, we should introduce some logic to take into consideration the fact that Stackage is not that fast.

### PR DESCRIPTION
**Pull request by v0d1ch**
This pr bumps the stack resolver to nightly-2018-10-01 which uses ghc 8.6.1

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
-resolver: lts-12.9
+resolver: nightly-2018-10-01
```
Comment (vrom911) [inline stack.yaml]: I guess you shouldn't change the `stack.yaml` file as we still want to support `8.4.3` instead you can put this into `stack-8.6.1.yaml` file 🙂

File: summoner.cabal [hunk 2]
```diff
@@ -47,7 +47,7 @@ library
   other-modules:       Paths_summoner
   autogen-modules:     Paths_summoner
 
-  build-depends:       base-noprelude >= 4.10 && < 5
+  build-depends:       base-noprelude >= 4.12 && < 5
```
Comment (vrom911) [inline summoner.cabal]: We are not going to give up the support of the `8.2.2` and `8.4.3` yet so no need to upgrade the lower bound. The cost of supporting those versions is very low (we already have everything working in there) but the usability on different versions is very valuable!

**Comment by v0d1ch** [comment]
@vrom911 I added the changes , It should be fine now.

File: stack-8.6.1.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+resolver: nightly-2018-10-01
+
+extra-deps:
+  - base-noprelude-4.12.0.0
+  - relude-0.3.0
```
Comment (ChShersh) [inline stack-8.6.1.yaml]: Actually, `relude-0.3.0` is already in this nightly snapshot, so you can safely remove it from `extra-deps`

Comment (v0d1ch) [reply stack-8.6.1.yaml]: Done!

**Pull request merged**